web/static/js/components/thesis-editor.js

```diff
@@ -223,6 +264,44 @@ class ThesisEditor extends React.Component {
     }
   }
 
+  importContent (t, d, page) {
+    switch (t) {
+      case 'image':
+      case 'background_image':
+        return this.imageEditor.uploadAndSet(d, page, () => { this.updateImportedContentCompletedCount() })
+      case 'text':
+        this.updateImportedContentCompletedCount()
+        return this.textEditor.set(d.name, d)
+      case 'html':
+        this.updateImportedContentCompletedCount()
+        return this.htmlEditor.set(d.name, d)
+      case 'raw_html':
+        this.updateImportedContentCompletedCount()
+        return this.rawHtmlEditor.set(d.name, d)
+      default:
+        return true
+    }
+  }
+
+  updateImportedContentCompletedCount () {
+    this.setState({importContentCompletedCount: ++this.state.importContentCompletedCount}, () => {
+      if (this.state.importContentCompletedCount === this.state.importContentQueueCount) {
```

Comment:
Any incrementing thing like this I always like to use `>==` just in case I somehow screw up and skip over the final number.